tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11236,1074 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+    VecInput1.push_back(NumberGenerator.generate());
+
+    if (IsVectorBinaryOp) {
+      VecInput2.push_back(NumberGenerator.generate());
+    }
+  }
+
+  // We pass these values into the shader and they're requried to compile. So
+  // they need to set to something.
+  T ClampArgC = 0;
+  T ClampArgT = 0;
+  if (TestConfig.IntrinsicString == "testClamp") {
+    ClampArgC = NumberGenerator.generate();
+    ClampArgT = NumberGenerator.generate();
+    while (ClampArgC >= ClampArgT) {
+      // Generate a new value for ClampArgC. It needs to be smaller than
+      // ClampArgT.
+      ClampArgT = NumberGenerator.generate();
+    }
+  }
+
+  std::vector<T> VecExpected = {};
+  for (size_t Index = 0; Index < VecInput1.size(); Index++) {
+    if (TestConfig.IsBinaryOp) {
+      T Input1 = VecInput1[Index];
+      T Input2 = TestConfig.IsScalarOp ? ScalarInput : VecInput2[Index];
+      if (TestConfig.OperatorString == "*")
+        VecExpected.push_back(Input1 * Input2);
+      else if (TestConfig.OperatorString == "+")
+        VecExpected.push_back(Input1 + Input2);
+      else if (TestConfig.OperatorString == ",") {
+        VERIFY_IS_TRUE(TestConfig.IntrinsicString != "",
+                       "Expecting intrinsic string");
+        if (TestConfig.IntrinsicString == "min")
+          VecExpected.push_back(std::min<T>(Input1, Input2));
+        else if (TestConfig.IntrinsicString == "max")
+          VecExpected.push_back(std::max<T>(Input1, Input2));
+        else
+          LogErrorFmt(L"Unrecognized BinaryOp intrinsic string: %s",
+                      TestConfig.IntrinsicString.c_str());
+      } else
+        LogErrorFmt(
+            L"Don't know how to compute expected value for operatorString: %s",
+            TestConfig.OperatorString.c_str());
+    } else // Unary op logic
+    {
+      if (TestConfig.IntrinsicString == "testClamp")
+        VecExpected.push_back(
+            std::clamp(VecInput1[Index], ClampArgC, ClampArgT));
+      else if (TestConfig.IntrinsicString == "testInitialize")
+        VecExpected.push_back(VecInput1[Index]);
+      else
+        LogErrorFmt(L"Unrecognized intrinsic string: %s",
+                    TestConfig.IntrinsicString.c_str());
+    }
+  }
+
+  // 1,2, and 3 are required to compile the shader.
+  // 1. -DOPERATOR : "*" "+" "," "-" "/" etc.
+  // 2. -DOPERAND2: : l.VecInput2 or scalarInput, depending on the test.
+  // 3. -DCLAMP_ARGC, -DCLAMP_ARG2 : clamp arg 1 and 2. Required to compile.
+  // 4. Optional: -DFUNC : min, max, etc.
+  std::string ShaderText = R"(
+  struct LongVectorOp{
+    TYPE ScalarInput;
+    vector<TYPE, NUM> VecInput1;
+    vector<TYPE, NUM> VecInput2;
+    vector<TYPE, NUM> VecOutput;
+  };
+
+  vector<TYPE, NUM> testInitialize(vector<TYPE, NUM> Vec)
+  {
+    vector<TYPE, NUM> VecCopy = Vec;
+    return VecCopy;
+  }
+
+  vector<TYPE, NUM> testClamp(vector<TYPE, NUM> Vec)
+  {
+    return clamp(Vec, TYPE(CLAMP_ARGC), TYPE(CLAMP_ARGT));
+  }
```

Comment:
As these aren't always used, suggest arranging for them to only be included in the shader if they're needed.